src/vm/tests/simple_apply_eval.rs

```diff
@@ -87,6 +87,31 @@ fn test_sha512() {
         .for_each(|(program, expectation)| assert_eq!(expectation, &p_to_hex(execute(program))));
 }
 
+#[test]
+fn test_sha512trunc256() {
+    let sha512_evals = [
+        "(sha512/256 \"\")",
+        "(sha512/256 0)",
+        "(sha512/256 \"The quick brown fox jumps over the lazy dog\")",
+    ];
+
+    fn p_to_hex(val: Value) -> String {
+        match val {
+            Value::Buffer(BuffData { data }) => to_hex(&data),
+            _ => panic!("Failed")
+        }
+    }
+
+    let expectations = [
+        "c672b8d1ef56ed28ab87c3622c5114069bdd3ad7b8f9737498d0c01ecef0967a",
+        "e41c9660b04714cdf7249f0fd6e6c5556f54a7e04d299958b69a877e0fada2fb",
```

Comment:
Not sure how you got this value?  I got `10baad1713566ac2333467bddb0597dec9066120dd72ac2dcb8394221dcbe43d` for 0.  Code:

```
h = hashlib.new('sha512-256')
h.update('00'.decode('hex'))
print h.hexdigest()
# prints "10baad1713566ac2333467bddb0597dec9066120dd72ac2dcb8394221dcbe43d"
```

Reply:
That's because the hash isn't of a 1-byte array -- it's the the 128-bit little-endian integer. See:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=48d0f8c77247ab30c841427bd7c1999a

Reply:
_(Not sure if this is the right place to surface this concern)_
I feel like we have surfaced a lot of idiomatic rust i128 type behaviors in Clarity without any explicit intention to do so. Maybe this is okay, but wouldn't that warrant something like a SIP-X stating that "default Clarity integers in many areas behave like Rust i128 integers" -- or something along those lines?

Reply:
I don't think we have -- the SIP already states that ints are 128-bit, and I don't think we're surfacing any behavior that's unusual for 128-bit integers.